coala_quickstart/generation/SettingsClass.py

```diff
@@ -0,0 +1,256 @@
+        The setting name as a string.
+    :return:
+        The value of type annotated or False.
+    """
+    return func.__annotations__[key] if key in func.__annotations__ else False
+
+
+def in_default_args(func, key):
+    """
+    Checks if a setting name as key is present in function
+    annotations.
+    :param func:
+        Function object.
+    :param key:
+        The setting name as a string.
+    :return:
+        True if key is present in args with default value else False.
+    """
+    return True if key in get_default_args(func) else False
+
+
+def in_all_args(func, key):
+    """
+    Checks if a setting name as key is present in function
+    annotations.
+    :param func:
+        Function object.
+    :param key:
+        The setting name as a string.
+    :return:
+        True if key is present in args of a function else False.
+    """
+    return True if key in get_all_args(func) else False
+
+
+class SettingTypes:
+
+    """
+    Categorizes the settings into Type bool and Type others
+    """
+
+    def __init__(self, settings, function, function_name, bear, trigger):
+        """
+        :param settings:
+            Either a dict of non-optional settings of the form:
+            {'setting_name': ('Description.', <class 'type'>),}
+            or
+            a dict of optional settings of the form:
+            {'setting_name': default_values,}
+        :param function:
+            The function object i.e. either the run() method or
+            the create_arguments() method of the bear.
+        :param function_name:
+            Name of the function, either 'run' or 'create_arguments'
+        :param bear:
+            The current bear object.
+        :param trigger:
+            String of value either 'optional' or 'non-optional'
+            depending on type of settings
+        """
+        self.settings_bool = []
+        self.settings_others = []
+        self.fillup_settings(function, settings, bear, trigger)
+
+    def fillup_settings(self, function, settings, bear, trigger):
+        """
+        Fill settings_bool and settings_others depending upon whether the
+        particular setting by the bear takes a bool value or any other.
+        :param function:
+            The function object i.e. either the run() method or
+            the create_arguments() method of the bear.
+        :param settings:
+            Either a dict of non-optional settings of the form:
+            {'setting_name': ('Description.', <class 'type'>),}
+            or
+            a dict of optional settings of the form:
+            {'setting_name': default_values,}
+        :param bear:
+            The current bear object.
+        :param trigger:
+            String of value either 'optional' or 'non-optional'
+            depending on type of settings
+        """
+        for key in settings:
+            if trigger == 'optional':
+                self.fillup_optional_settings(key, function)
+            else:
```

Comment:
this should be `elif trigger == 'non-optional'` and the else clause after that should raise a Value error with a proper message of Invalid trigger type.